modules/mongodb/connector.go

```diff
@@ -14,16 +15,28 @@ type connector interface {
 	listDatabaseNames() ([]string, error)
 	dbStats(databaseName string) (*dbStats, error)
 	isReplicaSet() bool
+	isMongos() bool
 	replSetGetStatus() (*replSetStatus, error)
+	shardNodes() (*shardNodesResult, error)
+	shardDatabasesPartitioning() (*partitionedResult, error)
+	shardCollectionsPartitioning() (*partitionedResult, error)
+	shardChunks() (map[string]int64, error)
 	initClient(uri string, timeout time.Duration) error
 	close() error
 }
 
-// mongoCollector interface that helps abstracting and mocking the database layer.
+// mongoCollector interface that helps to abstract and mock the database layer.
 type mongoCollector struct {
 	Client           *mongo.Client
 	Timeout          time.Duration
 	isReplicaSetFlag *bool
+	isMongosFlag     *bool
+	aggregationFunc  func(
+		client *mongo.Client,
+		ctx context.Context,
```

Comment:
Let's make "ctx" the first parameter here (usually that is the pattern)

Reply:
yes, I agree. Very good suggestion